django/db/models/fields/related.py

```diff
@@ -1586,7 +1586,15 @@ class ForeignObject(RelatedField):
 
     def __init__(self, to, from_fields, to_fields, rel=None, related_name=None,
             related_query_name=None, limit_choices_to=None, parent_link=False,
-            on_delete=CASCADE, swappable=True, **kwargs):
+            on_delete=None, swappable=True, **kwargs):
+
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required kwarg for %s in Django 2.0. "
+                "Set the attribute explicitly instead." % self.__class__.__name__,
```

Comment:
Well, there are a number of existing cases of docs URLs in user-facing messages (especially in migrations, but other places as well). I think there is a lot to recommend them from a UX perspective; someone who gets this warning is quite likely to in short order need a reference of the available `on_delete` options.

I don't think the URL-going-stale issue is that big for a deprecation warning, which has a limited lifespan anyway. If someone does happen to rearrange those docs in the next couple years, we'd  just update the message. (Like the ones in migrations, the message should use a Django-version-specific docs link, not the `stable` redirect, so that for a given Django version the link shouldn't ever go stale.)